lib/views/customer_details.erb

```diff
@@ -17,68 +17,209 @@
+                id="shipping-address-line1"
+                name="shipping_address_line1"
+                value="<%= @customer_details[:shipping_address_line1] %>"
+              />
+              <%if @errors.include?(:missing_shipping_address_line1)%>
+                <p class="text-danger">Missing Shipping Address</p>
+              <%end%>
             </div>
             <div class="form-group">
               <label for="shipping-address-line2">Address Line 2</label>
-              <input type="text" class="form-control" id="shipping-address-line2">
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-address-line2"
+                name="shipping_address_line2"
+                value="<%= @customer_details[:shipping_address_line2] %>"
+              />
             </div>
             <div class="form-group">
-              <label for="shipping-city">City</label>
-              <input type="text" class="form-control" id="shipping-city">
+              <label for="shipping-city">City<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-city"
+                name="shipping_city"
+                value="<%= @customer_details[:shipping_city] %>"
+              />
+              <%if @errors.include?(:missing_shipping_city)%>
+                <p class="text-danger">Missing Shipping City</p>
+              <%end%>
             </div>
             <div class="form-group">
-              <label for="shipping-county">County</label>
-              <input type="text" class="form-control" id="shipping-county">
+              <label for="shipping-county">County<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-county"
+                name="shipping_county"
+                value="<%= @customer_details[:shipping_county] %>"
+              />
+              <%if @errors.include?(:missing_shipping_county)%>
+                <p class="text-danger">Missing Shipping County</p>
+              <%end%>
             </div>
             <div class="form-group">
-              <label for="shipping-postcode">Postcode</label>
-              <input type="text" class="form-control" id="shipping-postcode">
+              <label for="shipping-postcode">Postcode<span class="text-danger">*</span></label>
+              <input
+                type="text"
+                class="form-control"
+                id="shipping-postcode"
+                name="shipping_postcode"
+                value="<%= @customer_details[:shipping_postcode] %>"
+              />
+              <%if @errors.include?(:missing_shipping_postcode)%>
+                <p class="text-danger">Missing Shipping Postcode</p>
+              <%elsif @errors.include?(:invalid_shipping_postcode)%>
```

Comment:
Change these error messages to be more user friendly. Eg  'Please enter a Shipping Postcode'